addons/sale_timesheet/report/project_profitability_report_analysis.py

```diff
@@ -30,189 +30,216 @@ class ProfitabilityAnalysis(models.Model):
-                       ELSE 0.0
-                    END AS amount_untaxed_to_invoice,
-                    CASE
-                       WHEN SOL.qty_delivered_method IN ('timesheet', 'manual') THEN (COALESCE(SOL.untaxed_amount_invoiced, COST_SUMMARY.downpayment_invoiced) / CASE COALESCE(S.currency_rate, 0) WHEN 0 THEN 1.0 ELSE S.currency_rate END)
-                       ELSE 0.0
-                    END AS amount_untaxed_invoiced,
-                    COST_SUMMARY.timesheet_unit_amount AS timesheet_unit_amount,
-                    COST_SUMMARY.timesheet_cost AS timesheet_cost,
-                    COST_SUMMARY.expense_cost AS expense_cost,
-                    COST_SUMMARY.other_revenues AS other_revenues
-                FROM project_project P
-                    JOIN res_company C ON C.id = P.company_id
-                    LEFT JOIN (
-                        SELECT
-                            project_id,
-                            analytic_account_id,
-                            sale_line_id,
-                            SUM(timesheet_unit_amount) AS timesheet_unit_amount,
-                            SUM(timesheet_cost) AS timesheet_cost,
-                            SUM(expense_cost) AS expense_cost,
-                            SUM(other_revenues) AS other_revenues,
-                            SUM(downpayment_invoiced) AS downpayment_invoiced
-                        FROM (
+                    sub.id as id,
+                    sub.project_id as project_id,
+                    sub.user_id as user_id,
+                    sub.sale_line_id as sale_line_id,
+                    sub.analytic_account_id as analytic_account_id,
+                    sub.partner_id as partner_id,
+                    sub.company_id as company_id,
+                    sub.currency_id as currency_id,
+                    sub.sale_order_id as sale_order_id,
+                    sub.order_confirmation_date as order_confirmation_date,
+                    sub.product_id as product_id,
+                    sub.sale_qty_delivered_method as sale_qty_delivered_method,
+                    sub.expense_amount_untaxed_to_invoice as expense_amount_untaxed_to_invoice,
+                    sub.expense_amount_untaxed_invoiced as expense_amount_untaxed_invoiced,
+                    sub.amount_untaxed_to_invoice as amount_untaxed_to_invoice,
+                    sub.amount_untaxed_invoiced as amount_untaxed_invoiced,
+                    sub.timesheet_unit_amount as timesheet_unit_amount,
+                    sub.timesheet_cost as timesheet_cost,
+                    sub.expense_cost as expense_cost,
+                    sub.other_revenues as other_revenues,
+                    (sub.expense_amount_untaxed_to_invoice + sub.expense_amount_untaxed_invoiced + sub.amount_untaxed_to_invoice +
+                        sub.amount_untaxed_invoiced + sub.other_revenues + sub.timesheet_cost + sub.expense_cost)
+                        as margin
```

Comment:
I assumed the margin would be computed as the difference of the absolute values of invoiced amounts and costs, and since `timesheet_cost` and `expense_cost` are already stored as negative values I added them instead.

Reply:
Oh yes. My bad. Well spotted ;)